src/Raven.Server/Documents/ShardedHandlers/ShardedSubscriptionsHandler.cs

```diff
@@ -0,0 +1,441 @@
+                    if (allSubs == null)
+                    {
+                        HttpContext.Response.StatusCode = (int)HttpStatusCode.NotFound;
+                        return;
+                    }
+
+                    if (running)
+                    {
+                        foreach (var sub in allSubs)
+                        {
+                            if (ShardedSubscriptionConnection.Connections.ContainsKey(sub.SubscriptionName))
+                            {
+                                subscriptions.Add(sub);
+                            }
+                        }
+                    }
+                    else
+                    {
+                        subscriptions = allSubs.ToList();
+                    }
+                }
+                else
+                {
+                    var subscription = id == null ? SubscriptionStorage.GetSubscriptionFromServerStore(ServerStore, context, ShardedContext.DatabaseName, name) : (SubscriptionStorage.SubscriptionGeneralDataAndStats)SubscriptionStorage.GetSubscriptionFromServerStoreById(ServerStore, ShardedContext.DatabaseName, id.Value);
+
+                    if (subscription == null)
+                    {
+                        HttpContext.Response.StatusCode = (int)HttpStatusCode.NotFound;
+                        return;
+                    }
+
+                    if (running)
+                    {
+                        if (ShardedSubscriptionConnection.Connections.ContainsKey(subscription.SubscriptionName) == false)
+                        {
+                            HttpContext.Response.StatusCode = (int)HttpStatusCode.NotFound;
+                            return;
+                        }
+                    }
+
+                    subscriptions = new List<SubscriptionStorage.SubscriptionGeneralDataAndStats> { subscription };
+                }
+
+                await using (var writer = new AsyncBlittableJsonTextWriter(context, ResponseBodyStream()))
+                {
+                    SubscriptionsHandler.WriteGetAllResult(writer, subscriptions, context);
+                }
+            }
+        }
+
+        [RavenShardedAction("/databases/*/subscriptions/drop", "POST")]
+        public async Task DropSubscriptionConnection()
+        {
+            var subscriptionId = GetLongQueryString("id", required: false);
+
+            var subscriptionName = GetStringQueryString("name", required: false);
+            if (subscriptionId.HasValue && string.IsNullOrEmpty(subscriptionName))
+                throw new InvalidOperationException("Drop Subscription Connection by id not supported for sharded connection.");
+
+            var workerId = GetStringQueryString("workerId", required: false);
+            try
+            {
+                using var result = await ShardedContext.TryExecuteCommandOnShards<object>(string.IsNullOrEmpty(workerId)
```

Comment:
this needs to be redesigned and match what we have currently with ShardExecutor

Reply:
done